nifi-nar-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/PutBoxFile.java

```diff
@@ -0,0 +1,366 @@
+        String folderId = context.getProperty(FOLDER_ID).evaluateAttributeExpressions(flowFile).getValue();
+        final String subfolderName = context.getProperty(SUBFOLDER_NAME).evaluateAttributeExpressions(flowFile).getValue();
+        final boolean createFolder = context.getProperty(CREATE_SUBFOLDER).asBoolean();
+        BoxFile.Info uploadedFile = null;
+
+        final long startNanos = System.nanoTime();
+
+        String fullPath = null;
+
+        try {
+            final long size = flowFile.getSize();
+
+            folderId = subfolderName != null ? getOrCreateParentSubfolder(subfolderName, folderId, createFolder).getID() : folderId;
+            final BoxFolder parentFolder = getFolder(folderId);
+            fullPath = BoxFileUtils.getPath(parentFolder.getInfo());
+
+            final long chunkUploadThreshold = context.getProperty(CHUNKED_UPLOAD_THRESHOLD)
+                    .asDataSize(DataUnit.B)
+                    .longValue();
+
+            final String conflictResolution = context.getProperty(CONFLICT_RESOLUTION).getValue();
+
+            try (InputStream rawIn = session.read(flowFile)){
+
+                boolean isNewVersionUpload = false;
+
+                if (REPLACE_RESOLUTION.equals(conflictResolution)) {
+                    final Optional<BoxFile.Info> alreadyUploadedFile = getFileByName(filename, parentFolder);
+
+                    if (alreadyUploadedFile.isPresent()) {
+                        BoxFile existingBoxFile = new BoxFile(boxAPIConnection, alreadyUploadedFile.get().getID());
+                        existingBoxFile.uploadNewVersion(rawIn);
+                        isNewVersionUpload = true;
+                    }
+                }
+
+                if (!isNewVersionUpload) {
+                    if (size > chunkUploadThreshold) {
+                        uploadedFile = parentFolder.uploadLargeFile(rawIn, filename, size);
+                    } else {
+                        uploadedFile = parentFolder.uploadFile(rawIn, filename);
+                    }
+                }
+
+            } catch (BoxAPIResponseException e) {
+                handleUploadError(conflictResolution, filename, parentFolder, e);
+            }
+
+            if (uploadedFile != null) {
+                final Map<String, String> attributes = BoxFileUtils.createAttributeMap(uploadedFile);
+                final String url = BOX_URL + uploadedFile.getID();
+                flowFile = session.putAllAttributes(flowFile, attributes);
+                final long transferMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
+                session.getProvenanceReporter().send(flowFile, url, transferMillis);
+            }
+
+            session.transfer(flowFile, REL_SUCCESS);
+        } catch (BoxAPIResponseException e) {
+            getLogger().error("Exception occurred while uploading file '{}' to Box folder '{}'", filename, fullPath, e);
```

Comment:
Recommend avoiding the wording `Exception occurred` and instead using something like the following. Also recommend including the Response Code for easier troubleshooting.
```suggestion
            getLogger().error("Upload failed: File [{}] Folder [{}] Response Code [{}]", filename, fullPath, e.getResponseCode(), e);
```